src/test/java/uk/gov/hmcts/reform/pip/data/management/service/SubscriptionManagementServiceTest.java

```diff
@@ -1,89 +1,53 @@
-import org.springframework.web.client.HttpServerErrorException;
-import org.springframework.web.client.RestTemplate;
 import uk.gov.hmcts.reform.pip.data.management.Application;
-import uk.gov.hmcts.reform.pip.data.management.config.AzureBlobConfigurationTest;
-import uk.gov.hmcts.reform.pip.data.management.config.RestTemplateConfigurationTest;
 import uk.gov.hmcts.reform.pip.data.management.models.publication.Artefact;
 
-import java.util.UUID;
-import java.util.concurrent.ConcurrentHashMap;
+import java.io.IOException;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.mockito.Mockito.doThrow;
-import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
+@ExtendWith(OutputCaptureExtension.class)
 @ActiveProfiles("test")
-@SpringBootTest(classes = {Application.class, AzureBlobConfigurationTest.class, RestTemplateConfigurationTest.class})
+@SpringBootTest(classes = {Application.class})
 @AutoConfigureEmbeddedDatabase(type = AutoConfigureEmbeddedDatabase.DatabaseType.POSTGRES)
 class SubscriptionManagementServiceTest {
 
-    @Autowired
-    RestTemplate restTemplate;
-
-    private static final String SUCCESS = "Success";
-    private static final UUID ARTEFACT_ID = UUID.randomUUID();
-    private static final String SHOULD_MATCH = "Lists should match";
-    private static final String SUBSCRIPTION_URL = "testUrl/subscription/artefact-recipients";
-    private final Artefact artefact = Artefact.builder()
-        .courtId("1")
-        .artefactId(ARTEFACT_ID)
-        .search(new ConcurrentHashMap<>())
+    private static final Artefact ARTEFACT = Artefact.builder()
+        .sourceArtefactId("TEST")
+        .provenance("PROVENANCE")
         .build();
+    private static MockWebServer mockSubscriptionManagementEndpoint;
 
-    @InjectMocks
     @Autowired
     SubscriptionManagementService subscriptionManagementService;
 
-    @BeforeEach
-    void setup() {
-        when(restTemplate.postForEntity(SUBSCRIPTION_URL, artefact, String.class))
-            .thenReturn(ResponseEntity.ok(SUCCESS));
-    }
-
-    @Test
-    void testSuccessReturnsList() {
-        assertEquals(SUCCESS, subscriptionManagementService.getSubscribersToArtefact(artefact),
-                     SHOULD_MATCH);
-    }
 
     @Test
-    void testEmptyListReturnedOnError() {
-        artefact.setCourtId("2");
-
-        HttpServerErrorException httpServerErrorException =
-            new HttpServerErrorException(HttpStatus.BAD_GATEWAY, "Bad Gateway", null, null);
-        doThrow(httpServerErrorException).when(restTemplate)
-            .postForEntity(SUBSCRIPTION_URL, artefact, String.class);
-
-        assertEquals("Subscription trigger unsuccessful for artefact: " + ARTEFACT_ID,
-                     subscriptionManagementService.getSubscribersToArtefact(artefact),
-                     SHOULD_MATCH);
+    void testSendTrigger() throws IOException {
+        mockSubscriptionManagementEndpoint = new MockWebServer();
```

Comment:
the docs are saying this is a integration test?

Reply:
https://www.baeldung.com/spring-mocking-webclient I went through this and it heavily recommended using mockwebserver vs traditional mocks (i.e. mockito)